### PR TITLE
(BOLT-126) Add WinRM Kerberos support

### PR DESCRIPTION
**Pull request by seanmil**
This is based on the work in #231, overhauled for the current version of Bolt.

### CODE REVIEW

**Comment by Iristyle** [comment]
Blocking on merge / release of WinRM gem based on bugfix at https://github.com/WinRb/WinRM/pull/302

**Comment by MikaelSmith** [comment]
I anticipate this will need some docs around setting it up as well. Who's going to take on writing them?

File: lib/bolt/transport/winrm/connection.rb [hunk 1]
```diff
@@ -31,20 +31,30 @@ def initialize(target, transport_logger)
         HTTPS_PORT = 5986
 
         def connect
-          if target.options['ssl']
+          if target.options['realm']
+            scheme = 'http'
+            transport = :kerberos
```
Comment (Iristyle) [inline lib/bolt/transport/winrm/connection.rb]: Will have to verify for sure, but Kerberos should actually work fine over HTTP or HTTPS, and we'll want to allow for both.

**Comment by Iristyle** [comment]
Closing in favor of #999 - thanks a ton for the initial work on this @seanmil !

**Pull request closed without merging**